pallene/coder.lua

```diff
@@ -1118,6 +1123,64 @@ gen_cmd["SetArr"] = function(self, cmd, _func)
     }))
 end
 
+gen_cmd["NewTable"] = function(self, cmd, _func)
+    local dst = self:c_var(cmd.dst)
+    local n = C.integer(cmd.size_hint)
+    return (util.render([[ $dst = pallene_new_table(L, $n); ]], {
+        dst = dst,
+        n = n,
+    }))
+end
+
+gen_cmd["GetTable"] = function(self, cmd, _func)
+    local dst = self:c_var(cmd.dst)
+    local tab = self:c_value(cmd.src_tab)
+    local key = self:c_value(cmd.src_k)
+    local dst_typ = cmd.dst_typ
+    local line = C.integer(cmd.loc.line)
+
+    return (util.render([[
+        {
+            static int cache = 0;
```

Comment:
It probably doesn't matter much, since the initial value of the cache gets quickly overwritten, but should this perhaps be initialized to -1 instead of 0?

Reply:
I don't see the point, this makes the code more complicated since -1 isn't a valid position. I think we should reword this variable as `size_t` to make the code clearer.

Reply:
If we choose to go the size_t route, -1 (aka UINT_MAX) could still be used as a marker of invalid cache position.